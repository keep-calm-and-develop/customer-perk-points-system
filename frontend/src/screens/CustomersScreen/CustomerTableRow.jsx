import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";

function CustomerTableRow({ name, customerID, totalPoints, monthlyPoints }) {
  return <tr data-testid="customer" className="customers-list__row">
        <td data-testid="customer-name" className="customers-list__cell-name">{name}</td>
        <td data-testid="customer-total-points" className="customers-list__cell-points">{totalPoints}</td>
        {
            monthlyPoints.map((points, idx) => (
                // eslint-disable-next-line react/no-array-index-key
                <td data-testid={`customer-month-${idx}-points`} className="customers-list__cell-points" key={`${customerID}-${idx}`}>{points}</td>
            ))
        }
        <td data-testid="customer-add-transaction"><Link className="customers-list__cell-add-transaction" to={`/${customerID}/new-transaction?name=${name}`}>Add Transaction</Link></td>
    </tr>
}

CustomerTableRow.propTypes = {
    name: PropTypes.string.isRequired,
    customerID: PropTypes.string.isRequired,
    totalPoints: PropTypes.number.isRequired,
    monthlyPoints: PropTypes.arrayOf(PropTypes.number).isRequired,
}

export default CustomerTableRow;