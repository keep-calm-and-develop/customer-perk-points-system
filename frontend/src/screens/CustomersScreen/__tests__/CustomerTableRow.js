import { cleanup, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import CustomerTableRow from "../CustomerTableRow";

afterEach(cleanup);

describe('<CustomerTableRow />', () => {
    it('Should match snapshot', () => {
        const customerData = {
            name: "user",
            customerID: '1234',
            totalPoints: 100,
            monthlyPoints: [50, 30, 20],
        };
        const customerTableRow = render(
            <BrowserRouter>
                <CustomerTableRow
                    name={customerData.name}
                    customerID={customerData.customerID}
                    totalPoints={customerData.totalPoints}
                    monthlyPoints={customerData.monthlyPoints}
                />
            </BrowserRouter>
        );
        expect(customerTableRow).toMatchSnapshot();
    });
    
    it(' Should render correct  ', async () => {
        const customerData = {
            name: "user",
            customerID: '1234',
            totalPoints: 100,
            monthlyPoints: [50, 30, 20],
        };
        await act(async () => render(
            <BrowserRouter>
                <CustomerTableRow
                    name={customerData.name}
                    customerID={customerData.customerID}
                    totalPoints={customerData.totalPoints}
                    monthlyPoints={customerData.monthlyPoints}
                />
            </BrowserRouter>
        ));
        const customerNameElm = await waitFor(() => screen.findByTestId("customer-name"));
        expect(customerNameElm).toHaveTextContent(customerData.name);
        const customerTotalPointsElm = await waitFor(() => screen.findByTestId("customer-total-points"));
        expect(customerTotalPointsElm).toHaveTextContent(customerData.totalPoints);
        const firstMonthPointsElm = await waitFor(() => screen.findByTestId("customer-month-0-points"));
        expect(firstMonthPointsElm).toHaveTextContent(customerData.monthlyPoints[0]);
        const secondMonthPointsElm = await waitFor(() => screen.findByTestId("customer-month-1-points"));
        expect(secondMonthPointsElm).toHaveTextContent(customerData.monthlyPoints[1]);
        const thirdMonthPointsElm = await waitFor(() => screen.findByTestId("customer-month-2-points"));
        expect(thirdMonthPointsElm).toHaveTextContent(customerData.monthlyPoints[2]);
    });
});