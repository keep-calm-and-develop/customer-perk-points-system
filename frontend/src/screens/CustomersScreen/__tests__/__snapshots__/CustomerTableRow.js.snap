// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerTableRow /> Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="customers-list__row"
        data-testid="customer"
      >
        <td
          class="customers-list__cell-name"
          data-testid="customer-name"
        >
          user
        </td>
        <td
          class="customers-list__cell-points"
          data-testid="customer-total-points"
        >
          100
        </td>
        <td
          class="customers-list__cell-points"
          data-testid="customer-month-0-points"
        >
          50
        </td>
        <td
          class="customers-list__cell-points"
          data-testid="customer-month-1-points"
        >
          30
        </td>
        <td
          class="customers-list__cell-points"
          data-testid="customer-month-2-points"
        >
          20
        </td>
        <td
          data-testid="customer-add-transaction"
        >
          <a
            class="customers-list__cell-add-transaction"
            href="/1234/new-transaction?name=user"
          >
            Add Transaction
          </a>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="customers-list__row"
      data-testid="customer"
    >
      <td
        class="customers-list__cell-name"
        data-testid="customer-name"
      >
        user
      </td>
      <td
        class="customers-list__cell-points"
        data-testid="customer-total-points"
      >
        100
      </td>
      <td
        class="customers-list__cell-points"
        data-testid="customer-month-0-points"
      >
        50
      </td>
      <td
        class="customers-list__cell-points"
        data-testid="customer-month-1-points"
      >
        30
      </td>
      <td
        class="customers-list__cell-points"
        data-testid="customer-month-2-points"
      >
        20
      </td>
      <td
        data-testid="customer-add-transaction"
      >
        <a
          class="customers-list__cell-add-transaction"
          href="/1234/new-transaction?name=user"
        >
          Add Transaction
        </a>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
